src/main/java/doggytalents/common/entity/ai/DogFollowOwnerGoal.java

```diff
@@ -98,9 +96,9 @@ public void tick() {
             this.timeToRecalcPath = 10;
             if (!this.dog.isLeashed() && !this.dog.isPassenger()) { // Is not leashed and is not a passenger
                 if (this.dog.distanceToSqr(this.owner) >= 144.0D) { // Further than 12 blocks away teleport
-                    EntityUtil.tryToTeleportNearEntity(this.dog, this.navigator, this.owner, 4);
+                    EntityUtil.tryToTeleportNearEntity(this.dog, this.dog.getNavigation(), this.owner, 4);
                 } else {
-                    this.navigator.moveTo(this.owner, this.followSpeed);
+                    this.dog.getNavigation().moveTo(this.owner, this.followSpeed);
```

Comment:
Make all dog::navigation references in DogFollowOwnerGoal dynamic, so active navigation objects always get selected